modules/bullet/collision_object_bullet.h

```diff
@@ -44,13 +44,35 @@ class btCollisionShape;
 
 class CollisionObjectBullet : public ShapeOwnerBullet {
 protected:
-	class HackedBasis : public Basis {
+	class EnhancedBasis : public Basis {
```

Comment:
You shouldn't need to do this workaround. Let me know if #10641 isn't enough for you, we can make additions to the Basis itself.

Reply:
In this moment your function does more work than I need since it handle rotation that I don't need. Thus I prefer to handle it in this way.

Reply:
But you're still using the rotation: it's the matrix you obtain using `p_global_transform.orthonormalized()`